src/Common/tests/System/Net/Http/Http2LoopbackConnection.cs

```diff
@@ -443,18 +450,19 @@ private static (int bytesConsumed, HttpHeaderData headerData) DecodeLiteralHeade
             }
 
             string value;
-            (bytesConsumed, value) = DecodeString(headerBlock.Slice(i));
+            bool valueCompressed;
+            (bytesConsumed, value, valueCompressed) = DecodeString(headerBlock.Slice(i));
             i += bytesConsumed;
 
-            return (i, new HttpHeaderData(name, value));
+            return (i, new HttpHeaderData(name, value, huffmanEncodedName: nameCompressed, huffmanEncodedValue: valueCompressed));
         }
 
         private static (int bytesConsumed, HttpHeaderData headerData) DecodeHeader(ReadOnlySpan<byte> headerBlock)
         {
             int i = 0;
 
             byte b = headerBlock[0];
-            if ((b & 0b10000000) != 0)
+            if ((b & 0b10000000) == 0b10000000)
```

Comment:
When the comparison is changed here, it should also be changed (to be in sync) in L342 (`bool isHuffmanCoded = (headerBlock[0] & 0b10000000) != 0;`).
Codegen should be the same for both variants (`test dl, 0x80`).

Or extract this check to a simple method (which will be inlined).